plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -314,7 +318,49 @@ private String getTargetCompatibility(Project project, SourceSet sourceSet) {
   private List<String> getCompilerArgs(Project project, SourceSet sourceSet) {
     JavaCompile javaCompile = getJavaCompileTask(project, sourceSet);
     if (javaCompile != null) {
-      return javaCompile.getOptions().getCompilerArgs();
+      CompileOptions options = javaCompile.getOptions();
+
+      try {
+        DefaultJavaCompileSpec specs = new DefaultJavaCompileSpec();
+        specs.setCompileOptions(options);
+
+        if (GradleVersion.current().compareTo(GradleVersion.version("6.6")) >= 0) {
+          if (javaCompile.getOptions().getRelease().isPresent()) {
```

Comment:
Since the api document says: `If set, it will take precedences over the getSourceCompatibility() and getTargetCompatibility() settings.`

what about extract the spec setup logics into a method, and make an early return when `release` is available? Then there is no need to check `specs.getRelease() == null` in the below two if blocks.